File: common/src/main/java/net/yourpackage/yourmod/ModCreativeTab.java
package net.yourpackage.yourmod;

import net.minecraft.core.Holder;
import net.minecraft.core.registries.BuiltInRegistries;
import net.minecraft.network.chat.Component;
import net.minecraft.resources.ResourceLocation;
import net.minecraft.world.item.CreativeModeTab;
import net.minecraft.world.item.ItemStack;
import net.minecraft.world.level.block.Blocks;

import static net.yourpackage.yourmod.MyMod.platform;

public class ModCreativeTab {

    public static final Holder<CreativeModeTab> EXAMPLE_CREATIVE_TAB = platform.register(
        BuiltInRegistries.CREATIVE_MODE_TAB,
        ResourceLocation.fromNamespaceAndPath(MyMod.modID, "my_creative_tab"),
        () -> platform.creativeTabBuilder()
            .icon(() -> new ItemStack(Blocks.GRASS_BLOCK))
            .title(Component.literal("My custom Creative Tab!"))
            .displayItems(((itemDisplayParameters, output) -> {
                for (var item : ModItems.ALL_ITEMS) output.accept(item.value());
            })).build()
    );

    public static void init() { }
}

File: common/src/main/java/net/yourpackage/yourmod/ModCreativeTab.kt
package net.yourpackage.yourmod

import net.minecraft.core.Holder
import net.minecraft.core.registries.BuiltInRegistries
import net.minecraft.network.chat.Component
import net.minecraft.resources.ResourceLocation
import net.minecraft.world.item.CreativeModeTab
import net.minecraft.world.item.Item
import net.minecraft.world.item.ItemStack
import net.minecraft.world.level.block.Blocks

@Suppress("unused")
object ModCreativeTab {

    val EXAMPLE_CREATIVE_TAB: Holder<CreativeModeTab> = MyMod.platform.register<CreativeModeTab>(
        BuiltInRegistries.CREATIVE_MODE_TAB,
        ResourceLocation.fromNamespaceAndPath(MyMod.modID, "my_creative_tab")
    ) {
        MyMod.platform.creativeTabBuilder()
            .icon { ItemStack(Blocks.GRASS_BLOCK) }
            .title(Component.literal("My custom Creative Tab!"))
            .displayItems { _: CreativeModeTab.ItemDisplayParameters?, output: CreativeModeTab.Output? ->
                for (item in ALL_ITEMS) output!!.accept(item.value())
            }.build()
    }

    val ALL_ITEMS: MutableList<Holder<Item>> = ArrayList()

    fun init() {}
}
